### PR TITLE
Miodrag/20190117 m71 python

### PR DESCRIPTION
**Pull request by MioMil**
Implemented/modified external audio device in m71.

### CODE REVIEW

**Review by robin-raymond — CHANGES_REQUESTED** [review]
This appears to be requesting to merge to master; we can't do that as it must be merged into releases/m71 and not direct to master.

**Comment by MioMil** [comment]
The only merge combination that exposes only the files I changed is the merge `Sergej/20181206-m71-python` because that was the base point to create my branch. I hope that this is correct choice for this pull-request.

**Review by robin-raymond — APPROVED** [review]
I'd like @morosev to review as well.

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.cc [hunk 1]
```diff
@@ -780,6 +780,8 @@ namespace webrtc
       inputDeviceIndex_(0),
       outputDeviceIndex_(0),
       newMicLevel_(0),
+		  recordingEnabled_(true),
+		  playoutEnabled_(true),
```
Comment (morosev) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.cc]: Tabs instead of spaces...

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.cc [hunk 2]
```diff
@@ -901,6 +903,9 @@ namespace webrtc
   void AudioDeviceWasapi::init(const CreationProperties &props) noexcept {
     id_ = String(props.id_);
 
+		recordingEnabled_ = props.recordingEnabled_;
+		playoutEnabled_ = props.playoutEnabled_;
+
```
Comment (morosev) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.cc:908]: Fix tabs.

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h [hunk 3]
```diff
@@ -512,6 +512,9 @@ namespace webrtc {
       uint16_t                                newMicLevel_;
 
       std::unique_ptr<DefaultAudioDeviceWatcher> defaultDeviceWatcher_;
+
+			bool                                    recordingEnabled_;
+			bool                                    playoutEnabled_;
```
Comment (morosev) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.h]: Tabs...

File: windows/wrapper/impl_webrtc_IAudioDeviceWasapi.h [hunk 4]
```diff
@@ -48,6 +48,9 @@ namespace webrtc
       IAudioDeviceWasapiDelegatePtr delegate_;
 
       const char *id_{};
+
+			bool recordingEnabled_;
+			bool playoutEnabled_;
```
Comment (morosev) [inline windows/wrapper/impl_webrtc_IAudioDeviceWasapi.h]: Already defined in inherited class AudioDeviceWasapi.

**Review by morosev — CHANGES_REQUESTED** [review]

**Review by morosev — APPROVED** [review]

**Pull request merged**